Automattic/wp-calypso
# Help Center: only check chat availability when user is eligible

#### Proposed Changes

* Currently, [we start a socket connection for every Help Center user to see if they have a chat session](https://github.com/Automattic/wp-calypso/blob/fix/change-message-when-continuing-session/packages/help-center/src/components/help-center-container.tsx#L36). We need to only do that for eligible users to save on Socket connections. All these connections are a waste when we don't render the chat option at all for ineligible users. This PR does that.

#### Testing Instructions
1. Start ETK by going in `apps/editing-toolkit` and running `yarn dev --sync`.
2. Open DevTools and go to the Network tab. Filter by WS. 
3. Using a free user, go to the editor of a sandboxed site.
4. You shouldn't see a WebSocket connection to Happychat server.
5. Repeat with a premium user, you should see a connection.
